source/Fleet.cpp

```diff
@@ -96,30 +96,35 @@ void Fleet::Enter(const System &system, list<shared_ptr<Ship>> &ships, const Pla
 {
 	if(!total || !government || variants.empty())
 		return;
-	
+
 	// Pick a fleet variant to instantiate.
 	const Variant &variant = ChooseVariant();
 	if(variant.ships.empty())
 		return;
-	
+
 	// Where this fleet can come from depends on whether it is friendly to any
 	// planets in this system and whether it has jump drives.
 	vector<const System *> linkVector;
 	// Find out what the "best" jump method the fleet has is. Assume that if the
 	// others don't have that jump method, they are being carried as fighters.
 	// That is, content creators should avoid creating fleets with a mix of jump
 	// drives and hyperdrives.
-	int jumpType = 0;
+	bool hasJumpDrive = false;
+	double jumpType = 0.;
```

Comment:
@EndrosG You're right, I completely missed the fact that I don't even need to get the cost :) I also made it so fleets can have multiple drives, but use another approach, which I think would also work without any additional code to handle fighters/drones.